src/main/kotlin/no/nav/k9/los/nyoppgavestyring/domeneadaptere/k9saktillos/K9SakTilLosAdapterTjeneste.kt

```diff
@@ -271,31 +285,154 @@ class K9SakTilLosAdapterTjeneste(
         OppgaveFeltverdiDto(
             nøkkel = "totrinnskontroll",
             verdi = event.aksjonspunktTilstander.filter { aksjonspunktTilstandDto ->
-                aksjonspunktTilstandDto.aksjonspunktKode.equals("5015") && aksjonspunktTilstandDto.status !in (listOf(AksjonspunktStatus.AVBRUTT))
+                aksjonspunktTilstandDto.aksjonspunktKode.equals("5015") && aksjonspunktTilstandDto.status !in (listOf(
+                    AksjonspunktStatus.AVBRUTT
+                ))
             }.isNotEmpty().toString()
         )
     ).filterNotNull().toMutableList()
 
-    private fun utledAvventerSaksbehandler(
-        harManueltAksjonspunkt: Boolean,
-        harAutopunkt: Boolean,
+    internal fun utledAvventerflagg(
+        behandlingSteg: String?,
+        åpneAksjonspunkter: List<AksjonspunktTilstandDto>,
         oppgaveFeltverdiDtos: MutableList<OppgaveFeltverdiDto>
     ) {
-        if (harManueltAksjonspunkt && !harAutopunkt) {
-            oppgaveFeltverdiDtos.add(
-                OppgaveFeltverdiDto(
-                    nøkkel = "avventerSaksbehandler",
-                    verdi = "true"
+        if (behandlingSteg.isNullOrEmpty()) {
+            if (åpneAksjonspunkter.isEmpty()) {
+                oppgaveFeltverdiDtos.addAll(avventerIngen())
+                return
+            } else {
+                throw IllegalStateException("Aktivt aksjonspunkt: ${åpneAksjonspunkter.first().aksjonspunktKode}, men ikke aktivt behandlingssteg")
```

Comment:
Flott, da kan du se bort fra kommentarene mine om å sjekke behandlingstatus -- for her sikrer du at det må være satt et aktivt aksjonspunkt.